main.go

```diff
@@ -138,7 +139,7 @@ func main() {
 
 	sm := operatormetrics.GenerateServiceMonitor(s)
 	err = mgr.GetClient().Create(context.TODO(), s)
-	if err != nil {
+	if err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
This makes is slightly misleading when we
```
log.Info("Created Service")
```
but I don't think it would cause confusion - maybe we could log that we `ensured metrics service exists` or something. Not a big deal for me

Reply:
Yeah +1 to cleaning up the logging in this function while we are here

Reply:
Agree! I've given that whole section a bit of a tidy-up now, thanks for calling it out.